daemons/controld/controld_te_events.c

```diff
@@ -177,17 +186,38 @@ update_failcount(xmlNode * event, const char *event_node_uuid, int rc,
 
         /* Update the fail count, if we're not ignoring failures */
         if (!ignore_failures) {
-            attr_name = pcmk__failcount_name(rsc_id, task, interval_ms);
-            update_attrd(on_uname, attr_name, value, NULL, is_remote_node);
-            free(attr_name);
+            fail_pair = calloc(1, sizeof(pcmk__attrd_query_pair_t));
+
+            fail_name = pcmk__failcount_name(rsc_id, task, interval_ms);
+            fail_pair->name = fail_name;
+            fail_pair->value = value;
+            fail_pair->node = on_uname;
+
+            attrs = g_list_prepend(attrs, fail_pair);
         }
 
         /* Update the last failure time (even if we're ignoring failures,
          * so that failure can still be detected and shown, e.g. by crm_mon)
          */
-        attr_name = pcmk__lastfailure_name(rsc_id, task, interval_ms);
-        update_attrd(on_uname, attr_name, now, NULL, is_remote_node);
-        free(attr_name);
+        last_pair = calloc(1, sizeof(pcmk__attrd_query_pair_t));
+
+        last_name = pcmk__lastfailure_name(rsc_id, task, interval_ms);
+        last_pair->name = last_name;
+        last_pair->value = now;
+        last_pair->node = on_uname;
+
+        attrs = g_list_prepend(attrs, last_pair);
+
+        pcmk__attrd_api_update_list(NULL, attrs, NULL, NULL, NULL, opts);
```

Comment:
We should reuse attrd_api, probably the best approach is to functionize the do_update block into controld_attrd.c